# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/OriginOperations.cpp
# Commit: 27b1f7f62a00
# Full Hash: 27b1f7f62a00d003b61e7d3265ec1c7587f221fa
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2025-06-11 18:38:51
# Regressor Bug: 1962133
# File Overlap Count: 10
# Overlapping Files: dom/quota/CommonMetadata.h, dom/quota/test/gtest/TestQuotaManager.cpp, dom/quota/OriginInfo.cpp, dom/quota/QuotaManager.h, dom/quota/test/gtest/QuotaManagerTestHelpers.cpp
# Description:
#   Bug 1962133 - QM: Introduce OriginStateMetadata struct for origin metadata header; r=dom-storage-reviewers,hsingh
#   
#   This patch introduces a new OriginStateMetadata struct to represent origin
#   state separately from the other OriginMetadata fields. This prepares the origin
#   metadata loading and saving code for adding a new field like the accessed flag.
# ==============================================================================

diff -r 2d9e785ad3e5 -r 27b1f7f62a00 dom/quota/OriginOperations.cpp
--- a/dom/quota/OriginOperations.cpp	Wed Jun 11 07:39:03 2025 +0000
+++ b/dom/quota/OriginOperations.cpp	Wed Jun 11 07:47:39 2025 +0000
@@ -3536,8 +3536,9 @@
         timestamp = PR_Now();
       }
 
-      FullOriginMetadata fullOriginMetadata =
-          FullOriginMetadata{originMetadata, /* aPersisted */ true, timestamp};
+      FullOriginMetadata fullOriginMetadata = FullOriginMetadata{
+          originMetadata,
+          OriginStateMetadata{timestamp, /* aPersisted */ true}};
 
       // Usually, infallible operations are placed after fallible ones.
       // However, since we lack atomic support for creating the origin
@@ -3567,7 +3568,8 @@
         // origin directory has been just created.
 
         FullOriginMetadata fullOriginMetadata = FullOriginMetadata{
-            originMetadata, /* aPersisted */ true, timestamp};
+            originMetadata,
+            OriginStateMetadata{timestamp, /* aPersisted */ true}};
 
         aQuotaManager.InitQuotaForOrigin(fullOriginMetadata, ClientUsageArray(),
                                          /* aUsageBytes */ 0);